docs/design/SYSTEM_DESIGN.md

```diff
@@ -6,101 +6,137 @@
-with information such as hostnames, protocol and ports stating which traffic flows are of interest.
-4. Application developers can now expose their APIs by configuring [HTTPRoute resources](https://gateway-api.sigs.k8s.io/concepts/api-overview/#httproute).
-These routes are attached to a specific [Gateway resource](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) allowing application traffic to reach
-the upstream application.
+Envoy Gateway is configured statically at runtime and the managed data plane is configured dynamically through
+Kubernetes resources, primarily [Gateway API][gw_api] objects.
+
+#### Static Configuration
+Static configuration is used to configure various internal aspects of Envoy Gateway at runtime, i.e. change the
+GatewayClass controllerName, configure a Provider, etc. Currently, Envoy Gateway only supports runtime configuration
+through a configuration file. See the [configuration](../CONFIG.md) guide for additional details.
+
+#### Dynamic Configuration
+Dynamic configuration is based on the concept of a declaring the desired state of the data plane and using
+reconciliation loops to drive the actual state toward the desired state. The desired state of the data plane is
+defined as Kubernetes resources that provide the following services:
+* Infrastructure Management- Manage the data plane infrastructure, i.e. deploy, upgrade, etc. This configuration is
+  expressed through [GatewayClass][gc] and [Gateway][gw] resources. A TBD [Custom Resource][cr] can be referenced by
+  `gatewayclass.spec.parametersRef` to modify data plane infrastructure default parameters,
+  e.g. expose Envoy network endpoints using a NodePort service instead of a LoadBalancer service.
+* Traffic Routing- Define how to handle application-level requests to backend services. For example, route all HTTP
+  requests for "www.example.com" to a backend service running a web server. This configuration is expressed through
+  [HTTPRoute][hroute] and [TLSRoute][troute] resources that match, filter, and route traffic to a [backend][be].
+  Although a backend can be any valid Kubernetes Group/Kind resource, Envoy Gateway only supports a [Service][svc]
+  reference.
 
 ### Components
 
-#### Config Sources
-This component is responsible for consuming the user configuration from various platforms. Data persistence should be tied to the specific config source’s capabilities. For e.g. in Kubernetes, the resources will persist in `etcd`, if using the `path-watcher`, the resources will persist in a file.
-
-##### Kubernetes
-It watches the Kubernetes API Server for resources, fetches them, and publishes it to the translators for further processing.
-
-##### Path Watcher
-It watches for file changes in a path, allowing the user to configure Envoy Gateway using resource configurations saved in a file or directory.
-
-##### Config Server
-This is a HTTP/gRPC Server allowing Envoy Gateway to be configured from a remote endpoint. 
+#### Provider
```

Comment:
Would it make sense to be a little more specific and call these "config providers" since "providers" is a very generic term?